File: src/main/java/me/miki/shindo/api/websocket/WsIdentity.java
package me.miki.shindo.api.websocket;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

/**
 * Representa a identidade atual do jogador usada para autenticação no gateway WebSocket.
 *
 * Esta classe é imutável e preparada para ser usada em logs/debugging.
 */
@Getter
@ToString
@AllArgsConstructor
public class WsIdentity {

    /**
     * UUID do jogador. Pode ser um UUID online (Mojang/Microsoft) ou um UUID offline
     * derivado do nickname.
     */
    private final String uuid;

    /**
     * Nome do jogador exibido no gateway.
     */
    private final String name;

    /**
     * Lista de roles enviadas ao gateway para resolução de permissões.
     */
    private final String[] roles;

    /**
     * Tipo de conta (LOCAL, MICROSOFT, OFFLINE, etc).
     */
    private final AccountType accountType;
}

File: src/main/kotlin/com/shindoclient/shindo/api/websocket/WsIdentity.kt
package com.shindoclient.shindo.api.websocket

data class WsIdentity(
    val uuid: String,
    val name: String,
    val roles: Array<String>?,
    val accountType: AccountType,
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is WsIdentity) return false
        if (uuid != other.uuid) return false
        if (name != other.name) return false
        if (accountType != other.accountType) return false
        if (roles != null) {
            if (other.roles == null) return false
            if (!roles.contentEquals(other.roles)) return false
        } else if (other.roles != null) {
            return false
        }
        return true
    }

    override fun hashCode(): Int {
        var result = uuid.hashCode()
        result = 31 * result + name.hashCode()
        result = 31 * result + (roles?.contentHashCode() ?: 0)
        result = 31 * result + accountType.hashCode()
        return result
    }
}
